python/tk_framework_desktopserver/shotgun/scripts/cache_commands.py

```diff
@@ -0,0 +1,125 @@
+# Copyright (c) 2017 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import sys
+import cPickle
+
+def cache(cache_file, data, base_configuration, hash_data, engine_name):
+    import sqlite3
+    import sgtk
+
+    entity = dict(type=data["entity_type"], id=data["entity_id"])
+    project_entity = dict(id=data["project_id"], type="Project")
+
+    # Setup the bootstrap manager.
+    toolkit_mgr = sgtk.bootstrap.ToolkitManager()
+    toolkit_mgr.plugin_id = "basic.shotgun.cache"
+    toolkit_mgr.base_configuration = base_configuration
+
+    pcs = toolkit_mgr.get_pipeline_configurations(
+        project=project_entity,
+    ) or [dict(id=None)]
+
+    for pc in pcs:
```

Comment:
Also, we didn't have a valid use case before for returning the path and descriptor fields back with get_pipeline_configurations so we didn't include them back then. Now that we do, maybe it would make sense to return those as well? This would avoid a call back to the server to resolve everything.

@manneohrstrom , this would mean that the entities would now have fields `id`, `type`, `name`, `descriptor`, `path` (as a `ShotgunPath`). Does this sound reasonable to guys?